app/models/host_status/execution_status.rb

```diff
@@ -37,6 +41,47 @@ def to_label(options = {})
     end
   end
 
+  class ExecutionTaskStatusMapper
+    attr_accessor :task
+
+    def self.sql_conditions_for(status)
+      status = STATUS_NAMES.key(status) if status.is_a?(String)
+
+      case status
+        when OK
+          [ "state = 'stopped' AND (result = 'success' OR result = 'warning')" ]
```

Comment:
oh, I'd consider this as a bug in foreman-tasks then, anyway moving warning to ERROR state for now because I'd expect "partial failed" being failed as well